File: grimdb/sdk/php/src/Models/SyncPeer.php
<?php declare(strict_types=1);
namespace Grimlocker\Models;

class SyncPeer
{
    public function __construct(
        public readonly string $deviceId  = '',
        public readonly string $host      = '',
        public readonly int    $port      = 0,
        public readonly int    $seenAt    = 0,
        public readonly bool   $reachable = true,
    ) {}

    public static function fromArray(array $data): self
    {
        return new self(
            $data['device_id'] ?? '',
            $data['host']      ?? '',
            $data['port']      ?? 0,
            $data['seen_at']   ?? 0,
            $data['reachable'] ?? true,
        );
    }
}

File: grimdb/sdk/swift/Sources/GrimlockerSDK/Models/SyncPeer.swift
import Foundation

public struct SyncPeer: Codable, Identifiable, Sendable {
    public var id: String { deviceId }
    public let deviceId: String
    public let host: String
    public let port: Int
    public let seenAt: Int64
    public let reachable: Bool?

    enum CodingKeys: String, CodingKey {
        case deviceId = "device_id"
        case host
        case port
        case seenAt = "seen_at"
        case reachable
    }
}
